File: kvx_flutter/lib/domain/repositories/buzzer_repository.dart
import '../entities/buzzer_detail.dart';

abstract class BuzzerRepository {
  Future<BuzzerDetail> load(String deviceId);
  Future<List<BuzzerSource>> loadLinkedPirs(String deviceId);
  Future<List<BuzzerMotionEvent>> loadHistory(String deviceId);
  Future<List<AvailableBuzzerPir>> loadAvailablePirs(String deviceId);
  Future<void> link(String deviceId, BuzzerLinkConfiguration configuration);
  Future<void> unlink(String deviceId, String pirId);
  Future<BuzzerTestReceipt> test(String deviceId);
}

File: kvx/Domain/Repositories/BuzzerRepository.swift
import Foundation

protocol BuzzerRepository {
    func detail(deviceID: String) async throws -> BuzzerDetail
    func linkedPIRs(deviceID: String) async throws -> [BuzzerSource]
    func history(deviceID: String) async throws -> [BuzzerMotionEvent]
    func availablePIRs(deviceID: String) async throws -> [AvailableBuzzerPIR]
    func link(deviceID: String, configuration: BuzzerLinkConfiguration) async throws
    func unlink(deviceID: String, pirID: String) async throws
    func test(deviceID: String) async throws -> BuzzerTestReceipt
}
